### ISSUE TITLE
couchdb provisioning error on AWS

### ISSUE THREAD

**Issue by xristy** [BuddhistDigitalResourceCenter/buda-base#2]
There's a couple of `curl` in the couchdb provisioner that reference port `13598`. On the local installs, Vagrantfile includes:

`config.vm.network :forwarded_port, guest: 13598, host:  5984 # couchdb`

but this is not feasible on the AWS provisioning.

I haven't looked at what the curls are doing so I don't know if the AWS instance is properly provisioned or not.

It would be nice if there's a way to configure couchdb during installation to run on some other port than `5984` so that AWS and local instances are all on the same non-standard port. The AWS security group would need to be updated if the port can be changed.

I've attached a segment from the provisioning transcript that shows the error.

[AWS-provisioning-error-couchdb.txt](https://github.com/BuddhistDigitalResourceCenter/buda-base/files/845934/AWS-provisioning-error-couchdb.txt)

**Comment by xristy** [comment]
I've looked further and I see that the couchdb on AWS and locally has the following ports in use:

> admin@ip-172-31-50-157:~$ sudo lsof -p 22929 |grep IPv4
beam.smp 22929 couchdb    8u  IPv4              35885      0t0    TCP *:58380 (LISTEN)
beam.smp 22929 couchdb    9u  IPv4              35887      0t0    TCP localhost:52875->localhost:epmd (ESTABLISHED)
beam.smp 22929 couchdb   14u  IPv4              35901      0t0    TCP localhost:5986 (LISTEN)
beam.smp 22929 couchdb   21u  IPv4              34922      0t0    TCP *:13598 (LISTEN)

I tried `13598` on AWS and on the locally provisioned buda-base and indeed it gives the expected response:

`{"couchdb":"Welcome","version":"2.0.0","vendor":{"name":"The Apache Software Foundation"}}`

I guess that you've already configured couchdb to listen on `13598` rather than its default of `5984`. 

I do see that same failure of `curl` on the local provisioning. So perhaps the curl is being issued too early.

I've changed the Vagrantfile port map:

`config.vm.network :forwarded_port, guest: 13598, host: 13598 # couchdb`

so the couchdb is can be contacted locally.

I've also updated the AWS security group to allow `13598`.

So there's still the failing curls during provisioning.

**Comment by eroux** [comment]
The curls just create the `test/` dataset and the design document, they fail if it already exists, so everything works as expected I think... just confirm that you're talking about the following:

```
{"error":"file_exists","reason":"The database could not be created, the file already exists."}
{"error":"conflict","reason":"Document update conflict."}
```

**Comment by xristy** [comment]
The included snippet from the provisioning shows:

`==> default: curl: (7) Failed to connect to localhost port 13598: Connection refused`

twice.

**Comment by xristy** [comment]
Élie solved the problem with [commit/48411986...](https://github.com/BuddhistDigitalResourceCenter/buda-base/commit/48411986b102806abf33b43bc1ba769b259b9128): `sleep 2` to allow couchdb time to startup fully
